rust/connlib/tunnel/src/client.rs

```diff
@@ -160,48 +165,45 @@ where
         Ok(())
     }
 
-    /// Sets the interface configuration and starts background tasks.
-    #[tracing::instrument(level = "trace", skip(self))]
-    pub fn set_interface(
-        &mut self,
-        config: &InterfaceConfig,
-        dns_mapping: BiMap<IpAddr, DnsServer>,
-    ) -> connlib_shared::Result<()> {
-        let device = Device::new(
-            config,
+    pub(crate) fn update_interface(&mut self) -> connlib_shared::Result<()> {
+        let dns_mapping = self.role_state.dns_mapping();
+        let mut device = Device::new(
+            self.role_state.interface_config.as_ref().expect("Developer error: we should always call update_interface after the interface config is set"),
             // We can just sort in here because sentinel ips are created in order
             dns_mapping.left_values().copied().sorted().collect(),
             self.callbacks(),
         )?;
 
+        device.set_routes(self.role_state.routes().collect(), &self.callbacks)?;
+
         self.device = Some(device);
         self.no_device_waker.wake();
```

Comment:
@conectado Do you think we can split creating the device from setting its IPs? I think that would simplify things within the tunnel because we would be able to get rid of the `Option`.

Reply:
We can't obtain a file descriptor in android until we have the config. We could make the `Option` internal to android.

Reply:
What I am looking for is a way of initializing the device without having to first talk to the portal. Having it an `Option` on Android only could work. We would return `Pending` until it is set and then call a waker.